### PR TITLE
build: bump sdk to v0.5.0

### LINKED ISSUE #57: Bump `stylus-sdk-rs` to v0.5.0

### PR DESCRIPTION
**Pull request by alexfertel**
Resolves #57 

Now that `nitro-testnode` has been updated, we are safe to bump the SDK version.

### CODE REVIEW

**Review by bidzyyys — COMMENTED** [review]
Looks good 🚀 
But E2E tests fail

File: .github/workflows/check-wasm.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
           toolchain: nightly-2024-01-01
```
Comment (qalisander) [inline .github/workflows/check-wasm.yml:48]: may be check out stable branch then?

**Review by qalisander — COMMENTED** [review]
Awesome that it works!

**Review by qalisander — APPROVED** [review]

File: .github/workflows/check-wasm.yml (hunk 1 above)
Comment (alexfertel) [reply .github/workflows/check-wasm.yml:48]: Yeah, let's try!

**Pull request merged**